### PR TITLE
Show macro name in 'this error originates in macro' message

### PR DESCRIPTION
**Pull request by Aaron1011**
When there are multiple macros in use, it can be difficult to tell
which one was responsible for producing an error.

### CODE REVIEW

File: src/test/ui/allocator/not-an-allocator.stderr
Comment (jyn514) [inline src/test/ui/allocator/not-an-allocator.stderr:8]: Unrelated to this PR, but would it be possible to only show "run with -Z macro-backtrace" in nightly builds, the same way the compiler only suggests `#![feature]` on nightly?

**Review by jyn514 — APPROVED** [review]

File: src/test/ui/allocator/not-an-allocator.stderr
Comment (camelid) [reply src/test/ui/allocator/not-an-allocator.stderr:8]: For months I've thought of suggesting this but kept forgetting, so thanks for suggesting it!

Comment (Aaron1011) [reply src/test/ui/allocator/not-an-allocator.stderr:8]: This can be a very useful option for debugging macros, so I think it's good to make people aware of it. Unlike `#![feature]`s, this option is something can be enabled just for local testing, without requiring that anyone else use a nightly compiler.

**Comment by Aaron1011** [comment]
This PR is extremely prone to bit-rotting. Once @matthewjasper or someone else is ready to r+ the general concept, I'll rebase and immediately approve it with p=1.

**Comment by JohnCSimon** [comment]
Triage:
Assigning back to author for build failure?

@Rustbot label: -S-waiting-on-review +S-waiting-on-author

**Comment by Aaron1011** [comment]
@JohnCSimon: The built failure is expected - I'll rebase and bless the tests once the overall concept is approved, due to the bitrot potentential.

File: src/test/ui/issues/issue-70724-add_type_neq_err_label-unwrap.stderr
Comment (cjgillot) [inline src/test/ui/issues/issue-70724-add_type_neq_err_label-unwrap.stderr:36]: Is `$crate` meant to be exposed this way?

**Comment by JohnCSimon** [comment]
ping from triage:
@Aaron1011 can you please address the build failures?
@Rustbot label: -S-waiting-on-review +S-waiting-on-author

**Comment by Aaron1011** [comment]
@JohnCSimon As I said before, this PR is waiting on review of the overall concept, since it's very prone to bitrot.

**Comment by JohnCSimon** [comment]
@Aaron1011 sorry, missed your previous comment

**Comment by Aaron1011** [comment]
@JohnCSimon No problem :)

File: compiler/rustc_errors/src/emitter.rs
Comment (estebank) [inline compiler/rustc_errors/src/emitter.rs:324]:
```suggestion
                    "this {level} originates in {descr} `{name}` \
```

**Review by estebank — APPROVED** [review]
r=me after rebasing

File: src/test/ui/allocator/not-an-allocator.stderr
Comment (estebank) [reply src/test/ui/allocator/not-an-allocator.stderr:8]: @Aaron1011's observation is exactly the reason I didn't gate that suggestion on being in nightly.

File: compiler/rustc_errors/src/emitter.rs
Comment (Aaron1011) [reply compiler/rustc_errors/src/emitter.rs:324]: I think the current output looks better - e.g. 'this error originates in the attribute macro `test`' vs 'this error originates in attribute macro `test`'

Comment (estebank) [reply compiler/rustc_errors/src/emitter.rs:327]: I'm not bothered either way :)

It seems like you have to rebless, but beyond that, r=me.

File: src/test/ui/issues/issue-70724-add_type_neq_err_label-unwrap.stderr
Comment (Aaron1011) [reply src/test/ui/issues/issue-70724-add_type_neq_err_label-unwrap.stderr]: Probably not. I'll open an issue to track resolving this.

**Pull request merged**
